modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/extension/IgniteTable.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.calcite.extension;
+
+import java.util.List;
+import org.apache.calcite.config.CalciteConnectionConfig;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelOptTable;
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.schema.Schema;
+import org.apache.calcite.schema.Statistic;
+import org.apache.calcite.schema.TranslatableTable;
+import org.apache.calcite.schema.Wrapper;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.util.ImmutableBitSet;
+import org.apache.ignite.internal.processors.query.calcite.schema.TableDescriptor;
+import org.apache.ignite.internal.processors.query.calcite.trait.IgniteDistribution;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Ignite table.
+ */
+public interface IgniteTable extends TranslatableTable, Wrapper {
+    /**
+     * Get table descriptor.
+     *
+     * @return Table descriptor.
+     */
+    TableDescriptor descriptor();
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    default RelDataType getRowType(RelDataTypeFactory typeFactory) {
```

Comment:
let's move all methods with default implementation to the bottom of the interface definition

Reply:
Unfortunately I can't, because it is overloaded method. 
"Placing non-overloaded methods in between overloaded methods with the same type is a violation."

Reply:
oh, ok then